Source/utils/parse_int.hpp

```diff
@@ -32,4 +36,30 @@ ParseIntResult<IntT> ParseInt(
 	return value;
 }
 
+inline uint8_t ParseFixed6Fraction(std::string_view str, const char **endOfParse = nullptr)

```

Comment:
"Base mana" can be negative or fractional, so it's probably not loaded correctly when it's negative with this code

Reply:
The integer part is loaded as a signed value, the fraction part is always positive though (it'll parse to a value in the range `[0, 1]`). I should add a test case to make sure since I don't think any of the existing test cases cover a class with negative starting/bonus mana.

Reply:
Oh I see, this only parses the fractional part. :+1: